### PR TITLE
2523-Use-NativeFileSystemBrowser-instead-of-just-NativeBrowser

### LINKED ISSUE #2523: Use NativeFileSystemBrowser instead of just NativeBrowser
As there could be different native browser types (file browser, web browser, ...)  can we please rename the new selectors like  #openInNativeBrowser into  #openInNativeFileSystemBrowser as this would be more clear.

### PR DESCRIPTION
**Pull request by hilaire**
Fixes #2523 
Use NativeFileSystemBrowser instead of just NativeBrowser

### CODE REVIEW

**Review by astares — APPROVED** [review]

**Review by jecisc — CHANGES_REQUESTED** [review]
This code is used in some projects. It would be better to use deprecation with transformation rules to not break them.

**Comment by astares** [comment]
@jecisc 
Do we really need the deprecation? To my knowledge the #openNativeBrowserOn was introduced in P8 and not earlier. 

Because it would mean to deprecate within a release iteration (P8 in this case) methods that were introduced within the same iteration...

**Comment by jecisc** [comment]
@astares It is used in Iceberg so without deprecation, it will break.

**Comment by astares** [comment]
@jecisc OK 

@hilaire can you please change the PR to
 - add a deprecation messages to #openNativeBrowserOn: and #openInNativeBrowser
    (or ideally a transformation rule so callers are changed automatically)
 - dispatch in the method to the new #openOSFileBrowserOn and #openInOSFileBrowser
 - move all the old methods to Deprecated80 package

File: src/Deprecated80/OSPlatform.extension.st [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Extension { #name : #OSPlatform }
+
+{ #category : #'*Deprecated80' }
+OSPlatform >> openNativeBrowserOn: aFileReference [ 
+	self deprecated: 'Use message openOSFileBrowserOn:'.
```
Comment (jecisc) [inline src/Deprecated80/OSPlatform.extension.st]:
```suggestion
	self deprecated: 'Use message openOSFileBrowserOn:' transformWith: '`@receiver openNativeBrowserOn: `@statements1' -> '`@receiver v `@statements1'.
```

File: src/Native-Browser/AbstractFileReference.extension.st [hunk 2]
```diff
@@ -2,5 +2,12 @@ Extension { #name : #AbstractFileReference }
 
 { #category : #'*Native-Browser' }
 AbstractFileReference >> openInNativeBrowser [
-	^ OSPlatform current openNativeBrowserOn: self
+	self deprecated: 'Use message openInOSFileBrowser'.
```
Comment (jecisc) [inline src/Native-Browser/AbstractFileReference.extension.st]:
```suggestion
	self deprecated: 'Use message openOSFileBrowserOn:' transformWith: '`@receiver openNativeBrowserOn: `@statements1' -> '`@receiver v `@statements1'.
```

**Review by jecisc — CHANGES_REQUESTED** [review]
I did suggestions to use rewrite rules for deprecation. They can be accepted directly via Github's web application.

File: src/Deprecated80/OSPlatform.extension.st (hunk 1 above)
Comment (hilaire) [reply src/Deprecated80/OSPlatform.extension.st]: Great. I did not know about transformation rule.

**Comment by hilaire** [comment]
I accepted your suggestions. Are we done with it?

**Comment by MarcusDenker** [comment]
errors not related

**Review by MarcusDenker — APPROVED** [review]

**Pull request merged**